src/esp/assets/ResourceManager.cpp

```diff
@@ -1985,6 +1989,112 @@ void ResourceManager::removePrimitiveMesh(int primitiveID) {
   primitive_meshes_.erase(primitiveID);
 }
 
+bool ResourceManager::importAsset(const std::string& filename,
+                                  bool requiresLighting) {
+  bool meshSuccess = true;
+  if (resourceDict_.count(filename) > 0) {
+    return true;
+  } else {
+    esp::assets::AssetInfo meshinfo{AssetType::UNKNOWN, filename};
+    meshinfo.requiresLighting = requiresLighting;
+    meshSuccess = loadRenderAssetGeneral(meshinfo);
+
+    // check if collision handle exists in collision mesh groups yet.  if not
+    // then instance
+    if (collisionMeshGroups_.count(filename) == 0) {
+      // set collision mesh data
+      const MeshMetaData& meshMetaData = getMeshMetaData(filename);
+
+      int start = meshMetaData.meshIndex.first;
+      int end = meshMetaData.meshIndex.second;
+      //! Gather mesh components for meshGroup data
+      std::vector<CollisionMeshData> meshGroup;
+      for (int mesh_i = start; mesh_i <= end; ++mesh_i) {
+        GenericMeshData& gltfMeshData =
+            dynamic_cast<GenericMeshData&>(*meshes_[mesh_i].get());
+        CollisionMeshData& meshData = gltfMeshData.getCollisionMeshData();
+        meshGroup.push_back(meshData);
+      }
+      collisionMeshGroups_.emplace(filename, meshGroup);
+    }
+  }
+
+  return meshSuccess;
+}
+
+std::string ResourceManager::setupMaterialModifiedAsset(
+    const std::string& filename,
+    const Magnum::Color4& matColor,
+    const Magnum::Color3& specularColor) {
+  assert(resourceDict_.count(filename) > 0);
+
+  std::string modifiedAssetName = filename + "_MatMod";
+
+  if (resourceDict_.count(modifiedAssetName) == 0) {
+    // first register the copied metaData
+    resourceDict_.emplace(modifiedAssetName,
+                          LoadedAssetData(resourceDict_.at(filename)));
+  }
+
+  // create/set a new PhongMaterialData for the asset
+  auto& meshMetaData = resourceDict_.at(modifiedAssetName).meshMetaData;
+
+  std::ostringstream matHandleStream;
+  matHandleStream << "phong_amb_" << matColor.toSrgbAlphaInt() << "_spec_"
```

Comment:
It would be really nice if this properties-to-string mapping went both ways. This would address my long-standing concerns about storing scene-specific info in ResourceManager.

As-is, a renderer in another process (such as a batched renderer) won't know what to do when it encounters a material called "my_material_phong_amb_01234567_spec_87654321" referenced by a SceneGraph/Drawable.